Code/GraphMol/FileParsers/MolSGroupWriting.cpp

```diff
@@ -636,6 +636,37 @@ std::string FormatV3000AttachPointBlock(
   return ret.str();
 }
 
+namespace {
+void addBlockToSGroupString(std::string block, std::string &currentLine,
+                            std::ostringstream &os) {
+  if (currentLine.length() + block.length() < 78) {
+    currentLine += block;
+  } else {
+    os << currentLine << " -\n";
+    if (block.length() < 73) {
+      os << "M  V30" << block << " -\n";
+      currentLine = "M  V30";
```

Comment:
So, if we have a long `currentLine` (e.g. 74 chars long), and a shot block (e.g., `MULT=3`), each of them would be written in a separate line, and we'd start with just the prefix as `currentLine`, e.g.:

```
M  V30 FIELDDATA="This is a very long property line that uses a line itself" -
M  V30 MULT=3 -
M  V30 [...]
```

I think we hit this in the tests you have added, but they don't check the actual output formatting, just that a roundtrip preserves the SGroups.

Maybe we could just remove this part and allow the other branch in the conditional to handle this too? I think the first conditional in this code will be enough to prevent RDKit from breaking lines in the middle of blocks.

Reply:
I don't believe this is true. I looked at the output molblocks in the tests before submitting the PR and they seem fine  - no empty `M  V30` lines.
I just confirmed this.

Reply:
Oh, true, I mixed up the contents of the lines.

But I still think both branches in this `if` statement could be replaced by the `while` loop and conditional after it.

Reply:
yep. It needed a bit of tweaking about the addition of white space, but it's now done.